### DISCUSSION TITLE
About Multiple JT Files importing in Java programmatically

### DISCUSSION THREAD

**Discussion by chinmayi1058** [Vertexvis/vertex-community#24] [Q&A] [answered]
How do I import multiple JT files of single assembly in Java programmatically to Vertexvis API?
This is the sample code by which I'm trying to import zip file of all necessary JT files to vertexvis through vertexvis API Client.

    `package com.vertexvis.example;

    import com.vertexvis.ApiClient;   //importing necessary jars of API from vertextvix
    import com.vertexvis.api.FilesApi;
    import com.vertexvis.api.StreamKeysApi;
    import com.vertexvis.model.CreateFileRequest;
    import com.vertexvis.model.CreateFileRequestData;
    import com.vertexvis.model.CreateFileRequestDataAttributes;
    import com.vertexvis.model.CreateStreamKeyRequest;
    import com.vertexvis.model.CreateStreamKeyRequestData;
    import com.vertexvis.model.CreateStreamKeyRequestDataAttributes;
    import com.vertexvis.model.FileMetadata;
    import com.vertexvis.model.Part;
    import com.vertexvis.model.Scene;
    import com.vertexvis.model.StreamKey;

    import java.io.File;
    import java.util.HashMap;
    import java.util.UUID;

    public class CreateSceneExample {
    public static void main(String[] args) throws InterruptedException {
	  //To authenticate registered user on Vertexvis client ID nad Secret ID required
    String id = "MY_CLIENT_ID"; //System.getenv("VERTEX_CLIENT_ID");
    String secret = "MY_SECRET_KEY"; //System.getenv("VERTEX_CLIENT_SECRET");
    if (isNullOrEmpty(id) || isNullOrEmpty(secret)) {
      throw new RuntimeException(    //if not provided will throw an exception
          "Environment variables containing your Vertex client ID and secret are required."
      );
    }

    try {
      ApiClient client =
          new ApiClient("https://platform.vertexvis.com", id, secret, new HashMap<>());     //constructor of APIClient class with this endpoint URL and authentication id and secret
     
      //PartCreator SceneCreator and FilesApi(to upload the JT file in vertexvis) each requires client 
      PartCreator partCreator = new PartCreator(client);
      SceneCreator sceneCreator = new SceneCreator(client);
      FilesApi files = new FilesApi(client);

      FileMetadata file = files.createFile(buildCreateFileReq("DemoEngine.zip"));    //creates an empty file with same name
      UUID fileId = file.getData().getId();   //to fetch unique fileid assigned to file from files metadata
      files.uploadFile(fileId, new File("C:\\Users\\ChinmayiAlshetty\\Downloads\\threeDVertex\\DemoEngine\\DemoEngine.zip"));

      Part part = partCreator.createPartFromFile(fileId);
      Scene scene = sceneCreator.createSceneFromPart(part);

      StreamKey sceneStreamKey =
          new StreamKeysApi(client).createSceneStreamKey(scene.getData().getId(),
              buildCreateStreamKeyReq());

      System.out.println(
          "Use key: '"
              + sceneStreamKey.getData().getAttributes().getKey()
              + "' to render scene: "
              + scene.getData().getId());
    } catch (RuntimeException e) {
      e.printStackTrace();
    }
    }

    private static CreateStreamKeyRequest buildCreateStreamKeyReq() {
      int TWENTY_FOUR_HOURS = 86400;
      return new CreateStreamKeyRequest()
        .data(
            new CreateStreamKeyRequestData()
                .type("stream-key")
                .attributes(
                    new CreateStreamKeyRequestDataAttributes().expiry(TWENTY_FOUR_HOURS)));   //expiry set in seconds 86400=24 hrs
    }

    private static CreateFileRequest buildCreateFileReq(String fileName) {
      return new CreateFileRequest()
        .data(
            new CreateFileRequestData()
                .type("file")
                .attributes(new CreateFileRequestDataAttributes().name(fileName)));
    }

    private static boolean isNullOrEmpty(String s) {
      return s == null || s.trim().length() == 0;
     }
    }
`
In DemoEngine folder all JT files are present along with the Jar file of all these JT files . This code should generate the respective Stream key of the scene after successful rendering done by vertexvis . But I'm getting the blank response . Let me know where I'm doing wrong.

**Comment by thomas-customer-success** [comment, marked as answer]
Hello, @chinmayi1058 .

The issue is most likely related to the input data `DemoEngine.zip`. If you are using the ZIP approach for importing a full assembly, there are a few considerations. First, ensure that the ZIP file includes all of the part files for the assembly. Also, please ensure that the new zipped filename matches the top-level assembly filename. For example, if your filename is `DemoEngine.zip`, there should be a filename like `DemoEngine.jt` within the ZIP which represents the top level assembly. Alternatively, you can also specify the filename of the top level assembly by using the `rootFileName` parameter as part of the [Create File Request](https://docs.vertexvis.com/#a8f05429-036c-45b8-89ef-6bd3aa14a7fe).

You might also find the [Developer Dashboard](https://dashboard.developer.vertexvis.com/) helpful during the troubleshooting process.

If you continue to run into problems feel free to reach out directly to support@vertexvis.com and we can continue to assist you.

Thomas
